doc/foreshadow_class.uml

```diff
@@ -0,0 +1,169 @@
+@startuml
+
+skinparam BackgroundColor transparent
+skinparam Shadowing false
+
+' Diagram setup
+hide empty members
+left to right direction
+set namespaceSeparator none
+
+package foreshadow.utils {
+    class check_df << (M,lemonchiffon) >>
+}
+package foreshadow.logging {
+    class ForeshadowLogger
+}
+package foreshadow.intents {
+    abstract class BaseIntent {
+        list engineering_pipeline
+        list preprocessing_pipeline
+        resolve_intent()
+    }
+
+    class DropIntent
+    class NumericalIntent
+    class CategoricalIntent
+    class TextIntent
+
+    BaseIntent <|-- DropIntent
+    BaseIntent <|-- NumericalIntent
+    BaseIntent <|-- CategoricalIntent
+    BaseIntent <|-- TextIntent
+
+    note "Config for intentless transformations (cleaner) are placed in\nBaseIntent's specification" as N1
+}
+
+together {
+    package foreshadow.transformers.core {
+        abstract class SmartTransformer {
+            bool fixed_fit
+            log_decision()
+        }
+
+        class ParallelProcessor
+        class SigCopy
+        class DropFeature
+
+        class wrap_transformer << (M,lemonchiffon) >>
+        wrap_transformer o-- SigCopy
+    }
+
+    package foreshadow.transformers.smart {
+        SmartTransformer <|-- Cleaner
+        SmartTransformer <|-- Engineerer
+        SmartTransformer <|-- Scaler
+        SmartTransformer <|-- Imputer
+        SmartTransformer <|-- CategoricalEncoder
+        SmartTransformer <|-- TextEncoder
+        SmartTransformer <|-- Reducer
+    }
+
+    package foreshadow.transformers.internal {
+        class FancyImpute
+        class UncommonRemover
+        class BoxCox
+
+        FancyImpute o-- Imputer
+        CategoricalEncoder o-- UncommonRemover
+        Scaler o-- BoxCox
+
+        class DaysSince
+        class NumericalFeatuerizer
+        class CategoricalFeatuerizer
+
+        Engineerer o-- DaysSince
+        Engineerer o-- NumericalFeatuerizer
+        Engineerer o-- CategoricalEncoder
+
+        class ToString
+        class SplitDate
+        class FinancialCleaner
+
+        Cleaner o-- ToString
+        Cleaner o-- SplitDate
+        Cleaner o-- FinancialCleaner
+
+        class Boruta
+        class Hypothesis
+
+        Reducer o-- Boruta
+        Reducer o-- Hypothesis
+    }
+
+    package foreshadow.transformers.external {
+        note "All sklearn transformers are mirrored\nand pandas wrapped here." as N3
+    }
+}
+
+package foreshadow.config {
+    class ConfigManager {
+        json framework_config
+        json user_config
+        json local_config
+    }
+}
+
+package foreshadow.tuners {
+    class WrappedTuner {
```

Comment:
We should call this TunerWrapper to be consistent with naming convention.
```suggestion
    class TunerWrapper {
```

Reply:
Sure.